### ISSUE TITLE
[BUG] Not expected behaviour with filter on percolate's _id

### ISSUE THREAD

**Issue by giovannialbero1992** [elastic/elasticsearch#19130]
**Elasticsearch version**:
2.3.3

**JVM version**:
java version "1.7.0_101"
OpenJDK Runtime Environment (IcedTea 2.6.6) (7u101-2.6.6-0ubuntu0.14.04.1)
OpenJDK 64-Bit Server VM (build 24.95-b01, mixed mode)

**OS version**:
Ubuntu 14.04.4

**Description of the problem including expected versus actual behavior**:
I've done a percolation on my index and i've applied a term filter on percolate's `_id` field, but the response is not the expected.

request
```
GET index/type/1/_percolate
{
  "filter": {
    "term": {
      "_id": "query_1"
    }
  }
}
```
expected response
```
{
  "took": 41,
  "_shards": {
    "total": 5,
    "successful": 5,
    "failed": 0
  },
  "total": 1,
  "matches": [
    {
      "_index": "index",
      "_id": "query_1"
    }
  ]
}
```

actual response
```
{
  "took": 14,
  "_shards": {
    "total": 5,
    "successful": 5,
    "failed": 0
  },
  "total": 0,
  "matches": []
}
```

**Steps to reproduce**:
 1. Create an index
 2. Store a query in percolator
 3. And do a percolation with filter on _id

**Comment by clintongormley** [comment]
Yes, for some reason the percolator filter doesn't support the `_id` lookup via `_uid`.  You can work around this by filtering on `_uid` instead:

    {
      "filter": {
        "term": {
          "_uid": ".percolator#query_1"
        }
      }
    }

Note: percolation has changed completely in 5.0.  Instead of the percolate API we now have the percolate query, which is just part of the query DSL.  As such, the rest of the query DSL will work as expected.

**Comment by martijnvg** [comment]
Closed via #19210
